src/celeritas/track/detail/TrackSortUtils.cc

```diff
@@ -43,6 +43,37 @@ void sort_impl(TrackSlots const& track_slots, F&& func, StreamId)
     std::sort(start, start + track_slots.size(), std::forward<F>(func));
 }
 
+// PRE: action_accessor is sorted, i.e. i <= j ==> action_accessor(i) <=
+// action_accessor(j)
+template<class F>
+void count_tracks_per_action_impl(Span<ThreadId> offsets,
+                                  size_type size,
+                                  F&& action_accessor)
```

Comment:
Name the function like a verb:
```suggestion
                                  F&& get_action)
```